bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManagerImpl.java

```diff
@@ -1262,6 +1256,10 @@ public Boolean isEnabled(ThingUID thingUID) {
         return thing.isEnabled();
     }
 
+    private boolean isDisabledByStorage(ThingUID thingUID) {
+        return storage == null || Boolean.FALSE.equals(storage.get(thingUID.getAsString()));
```

Comment:
this will return `true` in case no storage is defined. I think you want to say

```suggestion
        return storage != null && Boolean.FALSE.equals(storage.get(thingUID.getAsString()));
```